hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterFileSystem.java

```diff
@@ -382,4 +392,106 @@ public void stop() {
   public void logFileSystemState(Logger log) throws IOException {
     CommonFSUtils.logFileSystemState(fs, rootdir, log);
   }
+
+  // This API should be to be called on the 'old active cluster' before the switch over,
+  // to delete the associated active_cluster.id file.
+  public boolean prepareForSwitchOver() {
+    boolean status = true;
+    final Path idFile = new Path(rootdir, HConstants.ACTIVE_CLUSTER_SUFFIX_FILE_NAME);
+    try {
+      LOG.info("Deleting active cluster ID file");
+      FSUtils.removeFile(fs, idFile);
+    } catch (IOException ioe) {
+      LOG.warn("Failed to delete the active cluster ID file due to IOException", ioe);
+      status = false;
+    }
+    return status;
+  }
+
+  // This API should be called on the 'new active cluster' to complete the process of switch over,
+  // to create the new active_cluster.id file.
+  public boolean performSwitchOver() {
+    LOG.info("Creating new cluster ID file during switchover");
+    boolean status = true;
+    int wait = conf.getInt(HConstants.THREAD_WAKE_FREQUENCY, 10 * 1000);
+    try {
+      String ms = conf.get(HConstants.HBASE_META_TABLE_SUFFIX,
+        HConstants.HBASE_META_TABLE_SUFFIX_DEFAULT_VALUE);
+      if (
+        !FSUtils.checkFileExistsInHbaseRootDir(fs, rootdir,
+          HConstants.ACTIVE_CLUSTER_SUFFIX_FILE_NAME, wait)
+      ) {
+        FSUtils.setActiveClusterSuffix(fs, rootdir, new ActiveClusterSuffix(ms), wait);
+      }
+      activeClusterSuffix = FSUtils.getActiveClusterSuffix(fs, rootdir);
+    } catch (IOException ioe) {
+      // reset the cluster id
+      status = false;
+      LOG.error("Failed to create active cluster id file during switchover, exception", ioe);
+    }
+    return status;
+  }
+
+  private void negotiateActiveClusterIdFile(long wait) throws IOException {
+    LOG.info("Active cluster Suffix file");
+    String suffix_file_name = HConstants.ACTIVE_CLUSTER_SUFFIX_FILE_NAME;
+    boolean activeClusterSuffixFileExists =
+      FSUtils.checkFileExistsInHbaseRootDir(fs, rootdir, suffix_file_name, wait);
+
+    if (!isReadOnlyModeEnabled(conf)) {
+      // this is the active cluster, create active cluster suffix file if it does not exist
+      if (!activeClusterSuffixFileExists) {
+        FSUtils.setActiveClusterSuffix(fs, rootdir,
+          new ActiveClusterSuffix(getActiveClusterSuffixFromConfig(conf)), wait);
+      } else {
+        // verify the contents against the config set
+        ActiveClusterSuffix acs = FSUtils.getActiveClusterSuffix(fs, rootdir);
+        if (
+          acs.equals(getActiveClusterSuffix())
+            || (acs.toString().equals("null") && getActiveClusterSuffix().toString().isEmpty())
```

Comment:
We should add the clusterId to the file contents, in which case we could verify both: clusterId matches and suffix matches.